Topic1/src/com/topic1/MostExpensive.java

```diff
@@ -0,0 +1,31 @@
+package com.topic1;
+
+import java.util.HashSet;
+import java.util.Iterator;
+
+public class MostExpensive implements FindItem {
+
+	HashSet<Item> items;
+
+	public MostExpensive(HashSet<Item> items) {
+		this.items = items;
+
+	}
+
+	public Item find() {
+
+		double expensive = 0;
+		Item eItem = null;
+		Iterator<Item> it = items.iterator();
+		System.out.println(" " + (it.next()).getName()); // without this line
```

Comment:
[High] You should ask if *hasNext* item first.
Then you can store the next element in an auxiliar variable.